data-prepper-plugins/key-value-processor/src/main/java/com/amazon/dataprepper/plugins/processor/keyvalue/KeyValueProcessor.java

```diff
@@ -30,20 +30,40 @@
 
     private final KeyValueProcessorConfig keyValueProcessorConfig;
 
-    private final Pattern fieldDelimiterPattern;
+    private Pattern fieldDelimiterPattern = null;
```

Comment:
This should be `final`. If Java is unhappy about how you handle it in the constructor, create a temporary local variable. Then assign the field to that local variable.

Keeping fields final is very important to avoid errors since they can be used in any method.